src/shared/services/addresses/addresses.service.spec.ts

```diff
@@ -0,0 +1,91 @@
+
+    appConfig = jasmine.createSpyObj<AbstractAppConfig>('AppConfig', ['getPostcodeLookupUrl',
+      'getPostcodeLookupApiKey']);
+    httpService = jasmine.createSpyObj<HttpService>('HttpService', ['get']);
+    appConfig.getPostcodeLookupUrl.and.returnValue('http://postcodeUrl/postcode=${postcode}&key=${key}');
+    appConfig.getPostcodeLookupApiKey.and.returnValue('apikey');
+    let postCodeResponse = new Response(<ResponseOptions>{body: JSON.stringify(data)});
+    httpService.get.and.returnValue(Observable.of(postCodeResponse));
+    // httpService.get.and.returnValue(Observable.of([]));
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [AddressesService, {provide: AbstractAppConfig, useValue: appConfig}, {
+        provide: HttpService,
+        useValue: httpService
+      }]
+    });
+    injector = getTestBed();
+    addressesService = injector.get(AddressesService);
+    httpMock = injector.get(HttpTestingController);
+    httpTestingController = TestBed.get(HttpTestingController);
+  });
+
+  it('should be created with all the dependencies', () => {
+    expect(addressesService).toBeTruthy();
+  });
+
+  it('should return atleast more than one addresses from a given postcode', () => {
+    const result = addressesService.getAddressesForPostcode(validPostCode);
+    result.subscribe(i => { expect(i.length).toBeGreaterThan(2)});
+  });
+
+  it('should return all addresses from a given postcode location', () => {
+    const result = addressesService.getAddressesForPostcode(validPostCode);
+    result.subscribe(i => { expect(i.length).toEqual(12)});
+  });
+
+  it('should return addresses with either addressLine1 or addressLine2 populated', () => {
+    const result = addressesService.getAddressesForPostcode(validPostCode);
+    result.subscribe(addresses => {
+      expect(
+        addresses[0].AddressLine1.length > 0  || addresses[0].AddressLine2.length > 0
+      ).toBe(true);
+    });
+  });
+
+  it('should return addresses without null or undefined values ', () => {
+    const result = addressesService.getAddressesForPostcode(validPostCode);
+    result.subscribe(addresses => {
+      expect(addresses[0].AddressLine1).not.toContain("undefined");
+      expect(addresses[0].AddressLine1).not.toContain("null");
+      expect(addresses[0].AddressLine2).not.toContain("undefined");
+      expect(addresses[0].AddressLine2).not.toContain("null");
+    });
+
+  });
+
+  it('should return addresses with postcode value', () => {
+    const result = addressesService.getAddressesForPostcode(validPostCode);
+    result.subscribe(addresses => { expect(addresses[0].PostCode.length).toBeGreaterThan(0)});
+  });
+
+  it('should return subscriber error when postcode service returns zero', () => {
+    httpService.get.and.returnValue(Observable.of([]));
+    const result = addressesService.getAddressesForPostcode(validPostCode);
+    result.subscribe(addresses => expect(addresses.length).toBe(1),
```

Comment:
Why are we expecting addresses to be 1 if there is no values returned from `httpService`?

Reply:
This is a false assertion.